src/lib.rs

```diff
@@ -995,13 +956,9 @@ unsafe fn close_internal(c_out: &mut *mut Rav1dContext, flush: c_int) {
                     break;
                 }
                 pthread_join((*pf).task_thread.td.thread, 0 as *mut *mut c_void);
-                pthread_cond_destroy(&mut (*pf).task_thread.td.cond);
                 pthread_mutex_destroy(&mut (*pf).task_thread.td.lock);
                 n_0 = n_0.wrapping_add(1);
             }
-            pthread_cond_destroy(&mut (*ttd).delayed_fg.cond);
-            pthread_cond_destroy(&mut (*ttd).cond);
-            pthread_mutex_destroy(&mut (*ttd).lock);
```

Comment:
Where are these dropped?

Reply:
Or does it not matter at all since `{Mutex,Condvar}::new` are `const fn`s, which I think implies they don't have `Drop` `impl`s?

Reply:
It doesn't matter yet, there's no drop impl and we don't drop anything in the inner structs. Once we have proper ownership here it won't matter, so I don't want to add extra code that is a noop and will be removed.